src/reducers/alerts.js

```diff
@@ -1,34 +1,72 @@
+import alertsData from '../lib/alerts/index.jsx';
+import {AlertLevels} from '../lib/alerts/index.jsx';
 import extensionData from '../lib/libraries/extensions/index.jsx';
 
+const SHOW_STANDARD_ALERT = 'scratch-gui/alerts/SHOW_STANDARD_ALERT';
+const SHOW_EXTENSION_ALERT = 'scratch-gui/alerts/SHOW_EXTENSION_ALERT';
 const CLOSE_ALERT = 'scratch-gui/alerts/CLOSE_ALERT';
-const SHOW_ALERT = 'scratch-gui/alerts/SHOW_ALERT';
 
 const initialState = {
     visible: true,
+    // list of alerts, each with properties:
+    // * content (optional): react element (a <FormattedMessage />)
+    // * extentionId (optional): id string that identifies the extension
+    // * iconURL (optional): string
+    // * level (required): string, one of AlertLevels
+    // * message (optional): string
     alertsList: []
 };
 
 const reducer = function (state, action) {
     if (typeof state === 'undefined') state = initialState;
     switch (action.type) {
-    case SHOW_ALERT: {
+    case SHOW_STANDARD_ALERT: {
+        const alertId = action.alertId;
+        if (alertId) {
+            const newAlert = {
+                alertId: alertId,
+                level: AlertLevels.WARN // default level
+            };
+            const alertData = alertsData.find(thisAlertData => thisAlertData.alertId === alertId);
+            if (alertData) {
+                let newList = state.alertsList.slice();
+                newList = newList.filter(curAlert => (
```

Comment:
I'm concerned about not having the logic of this reducer in a unit test, would it be a lot to add a unit test for this reducer? Particularly covering the clearList logic?

Reply:
You're totally right! I added a handful of tests, including of this clearList logic.